python/ray/rllib/utils/vector_env.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import queue
+import threading
+
+
+class VectorEnv(object):
+    """An environment that supports batch evaluation."""
+
+    @staticmethod
+    def wrap(make_env=None, existing_envs=None, vector_width=1):
+        return _VectorizedGymEnv(make_env, existing_envs or [], vector_width)
+
+    @property
+    def action_space(self):
+        raise NotImplementedError
+
+    @property
+    def observation_space(self):
```

Comment:
If users are expected to extend VectorEnv, we should add documentation to each of these methods..